freyja/Dockerfile

```diff
@@ -1,27 +1,79 @@
-FROM mambaorg/micromamba:1.4
+FROM mambaorg/micromamba:2.0.2 as app
 
-ARG FREYJA_SOFTWARE_VERSION="1.4.7"
+# Version arguments
+# ARG variables only persist during build time
+# Bioconda Freyja version
+ARG FREYJA_BASE_SOFTWARE_VERSION="1.5.1"
+# Set if you want to override with a specific Github version tag
+ARG FREYJA_GITHUB_SOFTWARE_VERSION="1.5.2"
 
+# build and run as root users since micromamba image has 'mambauser' set as the $USER
 USER root
+# set workdir to default for building; set to /data at the end
 WORKDIR /
 
+LABEL base.image="mambaorg/micromamba:2.0.2"
+LABEL dockerfile.version="1"
+LABEL software="Freyja"
+LABEL software.version=${FREYJA_GITHUB_SOFTWARE_VERSION}
+LABEL description="Freyja is a tool to recover relative lineage abundances from mixed SARS-CoV-2 samples from a sequencing dataset (BAM aligned to the Hu-1 reference)"
+LABEL website="https://github.com/andersen-lab/Freyja"
+LABEL license="https://github.com/andersen-lab/Freyja/blob/main/LICENSE"
+
+
+# install dependencies; cleanup apt garbage
 RUN apt-get update && apt-get install -y --no-install-recommends \
     wget \
     ca-certificates \
     procps && \
     apt-get autoclean && rm -rf /var/lib/apt/lists/*
 
-RUN micromamba create -n freyja-env -c conda-forge -c bioconda -c defaults freyja=${FREYJA_SOFTWARE_VERSION} && \
-    micromamba clean -a -y
+# Create Freyja conda environment called freyja-env from bioconda recipe
+# clean up conda garbage
+# Install git to allow git clone
+RUN micromamba create -n freyja-env -c conda-forge -c bioconda -c defaults git freyja=${FREYJA_BASE_SOFTWARE_VERSION} && \
+  micromamba clean -a -y
 
+# set the environment, put new conda env in PATH by default
 ENV PATH="/opt/conda/envs/freyja-env/bin:/opt/conda/envs/env/bin:${PATH}" \
     LC_ALL=C.UTF-8
 
-RUN freyja update
+# Install freyja from Github source file
+RUN git clone --depth 1 --branch v${FREYJA_GITHUB_SOFTWARE_VERSION} https://github.com/andersen-lab/Freyja.git
```

Comment:
Something like the following would be smaller, but we still probably have to install git

```
RUN wget https://github.com/andersen-lab/Freyja/archive/refs/tags/v${FREYJA_GITHUB_SOFTWARE_VERSION}.tar.gz && \
pip install v${FREYJA_GITHUB_SOFTWARE_VERSION}.tar.gz && \
rm -rf v${FREYJA_GITHUB_SOFTWARE_VERSION}.tar.gz
```

Reply:
Yes, that would be another way of doing it and we will still need to install git. You can decide whichever way you prefer.